android/KMEA/app/src/main/assets/android-host.js

```diff
@@ -65,13 +93,21 @@ function notifyHost(event, params) {
 }
 
 // Update the KMW banner height
+// h is in dpi (different from iOS)
 function setBannerHeight(h) {
   if (h > 0) {
-    var osk = keyman.osk;
-    osk.banner.height = Math.ceil(h / window.devicePixelRatio);
+    // The banner itself may not be loaded yet.  This will preemptively help set
+    // its eventual display height.
+    bannerHeight = Math.ceil(h / window.devicePixelRatio);
+
+    if (keyman.osk) {
+      keyman.osk.bannerView.activeBannerHeight = bannerHeight;
+    }  
   }
-  // Refresh KMW OSK
+
+  // Refresh KMW's OSK
   keyman.refreshOskLayout();
+  doResetContext();
```

Comment:
Why do we need to reset the context when adjusting the banner?

Reply:
I got the pattern from ios-host.js

Reply:
reverted this and the other call too